Jenkinsfile

```diff
@@ -0,0 +1,134 @@
+pipeline {
+
+    agent {
+
+        label "pmem-csi"
+    }
+
+    options {
+
+        timeout(time: 2, unit: "HOURS")
+
+    }
+
+    environment {
+
+        PMEM_PATH = "/go/src/github.com/intel/pmem-csi"
+        IMAGE_VERSION = "ci-${env.BUILD_ID}"
+        BUILD_IMAGE = "clearlinux-builder"
+        CLUSTER= "ci-$IMAGE_VERSION"
+        GOVM_YAML="_work/$CLUSTER/deployment.yaml"
+        TEST_BOOTSTRAP_IMAGES="localhost:5000/pmem-csi-driver:$IMAGE_VERSION  localhost:5000/pmem-csi-driver-test:$IMAGE_VERSION"
+
+    }
+
+    stages {
+
+        stage('make test') {
+
+            options {
+
+                timeout(time: 60, unit: "MINUTES")
+                retry(3)
+
+            }
+
+            steps {
+
+                sh 'docker build --target build -t $BUILD_IMAGE .'
+                sh 'docker run --rm \
+                -v `pwd`:$PMEM_PATH $BUILD_IMAGE \
+                bash -c "ldconfig; make test"'
+
+            }
+
+        }
+
+        stage('Build test image') {
+
+            options {
+
+                timeout(time: 60, unit: "MINUTES")
+                retry(2)
+
+            }
+
+            steps {
+
+                sh 'docker run --rm \
+                    -e BUILD_IMAGE_ID=1 \
+                    -e IMAGE_VERSION=$IMAGE_VERSION \
+                    -v /var/run/docker.sock:/var/run/docker.sock \
+                    -v /usr/bin/docker:/usr/bin/docker \
+                    -v `pwd`:$PMEM_PATH \
+                    $BUILD_IMAGE \
+                    bash -c "make build-images"'
```

Comment:
This `make build-images` ends up rebuilding the build image because `CACHEBUST=1` here while it was the default value when building `BUILD_IMAGE` initially during the `make test` step.

If we set `BUILD_IMAGE_ID` consistently, ideally based on the current build number (`${env.BUILD_ID}`?), do we get Docker to reuse the layers?

Reply:
Setting `CACHEBUST=1` consistently was not enough to increase cache usage. Even during `make push-images` is the "swupd update" command run twice:
http://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/rest/organizations/jenkins/pipelines/pmem-CSI/branches/PR-305/runs/2/nodes/39/log/?start=0

That doesn't happen on my development machine (Docker 18.06.0-ce). The CI is using 18.09.6. Not sure where this different behavior is coming from.